File: NET/CDO/ActionRequest.cs
using System;
using System.Collections.Generic;
using System.Xml.Linq;
using System.Xml;
using System.Linq;

namespace CDO
{
	public class ActionRequest
	{
		public string ID = Guid.NewGuid().ToString();
		public string Name;
		public DateTime TimesOut;
		public DateTime AckTimesOut;
		public XElement Payload;

		public XElement GetXML()
		{
			XElement RVal = new XElement(CDO.Namespace + "actionrequest");
			RVal.SetAttributeValue("id", ID);
			RVal.SetAttributeValue("name", Name);
			RVal.SetAttributeValue("timeouts", XmlConvert.ToString(TimesOut, XmlDateTimeSerializationMode.Utc));
			RVal.SetAttributeValue("acktimeouts", XmlConvert.ToString(AckTimesOut, XmlDateTimeSerializationMode.Utc));

			RVal.Add(Payload);
			return RVal;
		}

		public ActionRequest()
		{
		}

		public ActionRequest(XElement Element)
		{
			this.Name = Element.Attribute("name").Value;
			this.ID = Element.Attribute("id").Value;
			this.TimesOut = XmlConvert.ToDateTime(Element.Attribute("timeouts").Value, XmlDateTimeSerializationMode.Utc);
			this.AckTimesOut = XmlConvert.ToDateTime(Element.Attribute("acktimeouts").Value, XmlDateTimeSerializationMode.Utc);
			this.Payload = Element.Elements().First();
		}

	}
}

File: typescript/CDO/ActionRequest.ts
var xmpp = require('../node-xmpp/packages/client').xmpp;
var xml = require('../node-xmpp/packages/xml');
var Guid = require('../Misc/Guid');
import {CDO} from './CDO';

export class ActionRequest 
{
    public ID:string = Guid.NewGuid();
    public Name:string;
    public TimesOut: Date;
    public AckTimesOut : Date;
    public Payload : any;

    public GetXML() : any
    {
        let RVal = xml("actionrequest", {xmlns:CDO.Namespace});
        RVal.attrs.id = this.ID;
        RVal.attrs.name = this.Name;
        RVal.attrs.timeouts = this.TimesOut.toISOString();
        RVal.attrs.acktimeouts = this.AckTimesOut.toISOString();

        RVal.append(this.Payload);
        return RVal;
    }

    public constructor(Data? : any)
    {
        if(Data)
        {
            this.Name = Data.attrs.name;
            this.ID = Data.attrs.id;
            this.TimesOut = new Date(Data.attrs.timesout);
            this.AckTimesOut = new Date(Data.attrs.acktimesout);
            this.Payload = Data.children[0];
        }
    }

}
